prefect_gcp/worker.py

```diff
@@ -582,6 +582,9 @@ async def run(
             CloudRunWorkerResult: A result object containing information about the
                 final state of the flow run
         """
+
+        self._run_logger = self.get_flow_run_logger(flow_run)
```

Comment:
It's not safe to assign to `self` like this — the worker instance is handling multiple flow runs at once